src/workerd/api/sockets.c++

```diff
@@ -143,7 +143,10 @@ jsg::Ref<Socket> setupSocket(
   if (!allowHalfOpen) {
     eofPromise = readable->onEof(js);
   }
-  auto writable = jsg::alloc<WritableStream>(ioContext, kj::mv(sysStreams.writable));
+  auto openedPrPair = js.newPromiseAndResolver<void>();
+  openedPrPair.promise.markAsHandled(js);
```

Comment:
What's the implication of marking this as handled? Does it suppress an exception from bubbling up if the promise is rejected?

On a related note, what happens if this promise is rejected now? It looks like `WritableStreamInternalController::closeImpl` won't be called? Is this OK because a `close()` would be a no-op anyway if the underlying stream failed to connect?

Reply:
> What's the implication of marking this as handled?

From v8 docs:

> Marks this promise as handled to avoid reporting unhandled rejections.

----

> On a related note, what happens if this promise is rejected now?

Good question. What will happen is that `close` will fail with the exception. I can see arguments why you may or may not want that to happen. But I think overall it shouldn't, since before this change you wouldn't get an exception. But I'd like to hear what @jasnell thinks before I change it.

Reply:
I've added a `catch_` to hopefully handle this case.

Reply:
> I've added a catch_ to hopefully handle this case.

Can we test it?

Reply:
Added test upstream